### DISCUSSION TITLE
Use CDN on library dependency

### DISCUSSION THREAD

**Discussion by felipe-chamas** [webpack/webpack#14168] [Q&A] [answered]
I'm currently with two projects bundled with Webpack:
- `Library`
- Main `Project`

The `Library` depends on [ThreeJS](threejs.org/), but I'd like to make it an external dependency loaded from CDN when I call `Library` inside `Project`.
I've created a sample repo\* to demonstrate the current behavior, with further information on the [README](https://github.com/FelipeR2U/external-three/blob/main/README.md), but the main issue is that the `Library` is working on itself and I can even test the CDN method with a `localhost` server. But when the `Project` imports it, I receive an error stating that `Library.function is not a function`
What may I be doing wrong?

\*Repo: https://github.com/FelipeR2U/external-three

**Comment by alexander-akait** [comment, marked as answer]
Why do you bundle library? ECMA modules supported by Node.js and not only, I recommend to avoid bundle libraries, anyway based on your example:
1. `public/index.html` has wrong `<script src="./dist/index.js"></script>` (change on ` <script src="./index.js"></script>`, but I think your generate them, so I just missing something in your example)
2. Change code on:
```
import * as mod from 'library'

(async () => {
    const test = await mod;
    
    test.createScene();
})();
```

Firstly - `umd` doesn't have `named` exports, 
Secondly - when you require an async module you will get a promise. You module is async due `external` logic to load `three.js`.

**Reply by alexander-akait** [reply]
Feel free to feedback

**Reply by felipe-chamas** [reply]
Woow, it actually worked!!! Thaks a lot. I will try it on my real project now.
Just to clarify your questions:

> Why do you bundle library? ECMA modules supported by Node.js and not only, I recommend to avoid bundle libraries

The Library is used by Project, and Project is also a library used by our clients on their website. So it’s better for them that our solution lazy-loads after Three has loaded than the whole website waiting for the entire project to show first content.
I’ve tried code-splitting in the past with those two, but since it’s not for my website, it’s quite hard to accomplish what we need.
(Also, the clients use our library from the window, I haven’t found any way of lazy-loading/code-splitting a window library on webpack…)

> 1.

I made it like this on Webpack:
```js
    output: {
      path: path.resolve(__dirname, 'public/dist'),
      filename: 'index.js',
      libraryTarget: 'umd'
    },
```
